core/arch/arm/mm/core_mmu.c

```diff
@@ -222,6 +226,87 @@ static struct tee_mmap_region *find_map_by_pa(unsigned long pa)
 	return NULL;
 }
 
+extern const struct core_mmu_phys_mem __start_phys_sdp_mem_section;
+extern const struct core_mmu_phys_mem __end_phys_sdp_mem_section;
+
+static bool pbuf_is_sdp_mem(paddr_t pbuf, size_t len)
+{
+	const struct core_mmu_phys_mem *mem;
+
+	for (mem = &__start_phys_sdp_mem_section;
+	     mem < &__end_phys_sdp_mem_section; mem++)
+		if (core_is_buffer_inside(pbuf, len, mem->addr, mem->size))
+			return true;
+
+	return false;
+}
+
+static bool pbuf_intersect_sdp_mem(paddr_t pbuf, size_t len)
+{
+	const struct core_mmu_phys_mem *mem;
+
+	for (mem = &__start_phys_sdp_mem_section;
+	     mem < &__end_phys_sdp_mem_section; mem++)
+		if (core_is_buffer_intersect(pbuf, len, mem->addr, mem->size))
+			return true;
+
+	return false;
+}
+
+static bool verify_sdp_mem_areas(void)
+{
+	const struct core_mmu_phys_mem *mem1;
+	const struct core_mmu_phys_mem *mem2;
+	const struct core_mmu_phys_mem *mem_start = &__start_phys_sdp_mem_section;
+	const struct core_mmu_phys_mem *mem_end = &__end_phys_sdp_mem_section;
+
+	for (mem1 = mem_start; mem1 < mem_end; mem1++)
```

Comment:
Maybe worth adding a comment here something like
```
/* Checking for overlaps between adjacent SDP memory areas */
```
Or even as a function comment explain what it actually is verifying.